src/main/java/com/hierynomus/mssmb2/messages/SMB2ChangeNotifyResponse.java

```diff
@@ -72,8 +72,12 @@ protected boolean isSuccess(NtStatus status) {
             fileNameLen = buffer.readUInt32();
             fileName = buffer.readString(StandardCharsets.UTF_16LE, (int) fileNameLen / 2);
             notifyInfoList.add(new FileNotifyInfo(action, fileName));
-            currentPos += nextEntryOffset;
-            buffer.rpos(currentPos);
+            if (nextEntryOffset != 0) {
+                currentPos += nextEntryOffset;
+                buffer.rpos(currentPos);
+            } else {
```

Comment:
There is no need for the else. After reading the last fileName the buffer is at the right position.